kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/StreamableKafkaMessageSource.java

```diff
@@ -153,12 +155,21 @@ private String buildConsumerGroupId() {
         private Supplier<String> groupIdSuffixFactory = () -> UUID.randomUUID().toString();
         private ConsumerFactory<K, V> consumerFactory;
         private Fetcher<K, V, KafkaEventMessage> fetcher;
-        @SuppressWarnings({"unchecked", "squid:S1874"})
-        private KafkaMessageConverter<K, V> messageConverter =
-                (KafkaMessageConverter<K, V>) DefaultKafkaMessageConverter.builder()
-                                                                          .serializer(XStreamSerializer.defaultSerializer())
-                                                                          .build();
+        private KafkaMessageConverter<K, V> messageConverter;
         private Supplier<Buffer<KafkaEventMessage>> bufferFactory = SortedKafkaMessageBuffer::new;
+        private Supplier<Serializer> serializer;
```

Comment:
I think it would be better to just set the default here already. Then we also don't need to check for null, the XStream serializer will already log if the code is called.